configure.ac

```diff
@@ -305,6 +306,42 @@ AC_CHECK_HEADER(valgrind/memcheck.h, [],
     AC_MSG_ERROR([valgrind requested but <valgrind/memcheck.h> not found.]))
 fi
 
+dnl Check for userfaultfd support
+have_userfaultfd=0
+AC_CHECK_HEADERS([linux/userfaultfd.h],
+    [AC_MSG_CHECKING([for userfaultfd interface])
+     AC_TRY_LINK([#include <sys/types.h>
+#include <linux/userfaultfd.h>
+#include <unistd.h>
+#include <sys/syscall.h>
+#include <fcntl.h>
+],
+        [return syscall(__NR_userfaultfd, O_CLOEXEC | O_NONBLOCK);],
+            [AC_MSG_RESULT([yes])
+             have_userfaultfd=1],
+            [AC_MSG_RESULT([no])
+             have_userfaultfd=0])])
+AS_IF([test "$have_userfaultfd" -eq "1"]
+    [AC_MSG_CHECKING([for userfaultfd unmap support])
+     AC_TRY_LINK([#include <sys/types.h>
```

Comment:
This one can be AC_COMPILE_IFELSE, but probably don't want a CHECK_DECL.  You'd need to check a bunch of different declarations, and then OR the results, and that doesn't seem useful.